core/modules/mlfqueue.cc

```diff
@@ -0,0 +1,346 @@
+#include "mlfqueue.h"
+#include "../mem_alloc.h"
+
+#include <rte_ether.h>
+#include <rte_ip.h>
+#include <rte_udp.h>
+#include <time.h>
+
+#include <glog/logging.h>
+
+#include <string>
+#include <cmath>
+#include <iostream>
+#include <fstream>
+
+bool operator==(const MLFQueue::Q_Id &id1, const MLFQueue::Q_Id &id2) {
+  bool ips = id1.src_ip == id2.src_ip && id1.dst_ip == id2.dst_ip;
+  bool ports = id1.src_port == id2.src_port && id1.dst_port == id2.dst_port;
+  return ips && ports && id1.protocol == id2.protocol;
+}
+bool operator<(const MLFQueue::Q_Id &id1, const MLFQueue::Q_Id &id2)  {
+  bool ips = id1.src_ip < id2.src_ip && id1.dst_ip < id2.dst_ip;
+  bool ports = id1.src_port < id2.src_port && id1.dst_port < id2.dst_port;
+  return ips && ports && id1.protocol < id2.protocol;
+}
+
+int RoundToPowerTwo(int val) {
+ return pow(2, ceil(log2(val)));
+}
+
+const Commands MLFQueue::cmds = {
+    {"set_num_lvls", "MlfqLvlArg", MODULE_CMD_FUNC(&MLFQueue::CommandNumPriorityLevels), 0},
+    {"set_batch_Size", "MlfqBatchArg", MODULE_CMD_FUNC(&MLFQueue::CommandBatchSize), 0},
+    {"set_max_flow_queue_size", "MlfqMaxFlowQueueSizeArg",
+          MODULE_CMD_FUNC(&MLFQueue::CommandMaxFlowQueueSize), 0}};
+
+pb_error_t MLFQueue::Init(const bess::pb::MlfqArg &arg) {
+  pb_error_t err;
+
+  if (arg.num_lvls() != 0) {
+    err = SetNumPriorityLevels(arg.num_lvls());
```

Comment:
Better to avoid duplicated code like this.  Instead, you could have (suitably modified for each instance, and for the pending change of constant name):

auto levels = arg.num_lvls() ? arg_num_lvls() : DEFAULT_NUM_LVLS;

Then you don't need the if statement or duplicated code.  Same for the other args.